src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/SubscriptionRouter.cs

```diff
@@ -15,19 +14,11 @@ public SubscriptionRouter(Publishers publishers, EndpointInstances endpointInsta
             this.transportAddressTranslation = transportAddressTranslation;
         }
 
-        public async Task<IEnumerable<string>> GetAddressesForEventType(Type messageType)
+        public IEnumerable<string> GetAddressesForEventType(Type messageType)
         {
-            var results = new HashSet<string>();
-            foreach (var publisherAddress in publishers.GetPublisherFor(messageType))
-            {
-                var addresses = await publisherAddress.Resolve(endpoint => endpointInstances.FindInstances(endpoint), i => transportAddressTranslation(i)).ConfigureAwait(false);
-                foreach (var address in addresses)
-                {
-                    results.Add(address);
-                }
-            }
-
-            return results;
+            var publisher = publishers.GetPublisherFor(messageType);
```

Comment:
Good catch. Although I think on the doco level we are clear that for each even there should be a single logical publisher, I am not sure how it should work on the code level. If we are now checking only concrete routes, without any inheritence, maybe we can use a single publisher?